### DISCUSSION TITLE
⬇️ [HOWTO]: Downloading grok-1 model | 下载 grok-1 模型

### DISCUSSION THREAD

**Discussion by trholding** [xai-org/grok-1#129] [Q&A] [answered]
Official Sources:

1. Torrent: https://academictorrents.com/details/5f96d43576e3d386c9ba65b883210a393b68210e
2. Magnet Link: 
```magnet:?xt=urn:btih:5f96d43576e3d386c9ba65b883210a393b68210e&tr=https%3A%2F%2Facademictorrents.com%2Fannounce.php&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce```
3. Hugging Face: https://huggingface.co/xai-org/grok-1 / https://huggingface.co/xai-org/grok-1/tree/main/ckpt-0

Quick way to download from Hugging Face:

```
huggingface-cli download xai-org/grok-1 --repo-type model --include ckpt-0/* --local-dir checkpoints --local-dir-use-symlinks False
```

If you get errors on zsh, then add quotes to ckpt-0/* to prevent shell from interpreting * as wildcard:

```
huggingface-cli download xai-org/grok-1 --repo-type model --include 'ckpt-0/*' --local-dir checkpoints --local-dir-use-symlinks False
```

Related to: https://github.com/xai-org/grok-1/discussions/91
Added information from: https://github.com/xai-org/grok-1/discussions/129#discussioncomment-8831561 https://github.com/xai-org/grok-1/issues/181#issuecomment-2005771207 https://github.com/xai-org/grok-1/issues/203#issuecomment-2007003097

**Comment by trholding** [comment, marked as answer]
Usage:

```
git clone https://github.com/xai-org/grok-1.git
cd grok-1/checkpoints
```

Then download the checkpoints via torrent or from hugginface as in https://github.com/xai-org/grok-1/discussions/129 and place ckpt directory (https://huggingface.co/xai-org/grok-1/tree/main/ckpt-0) into grok-1/checkpoints

Then do:

```
cd ..
pip install -r requirements.txt
python run.py
```

**Reply by breakinghong** [reply]
one question that why files from the 3 ways（torrent,magnet link,hugging face） not the same size？that makes me confused，I actually download by magnet link

**Reply by traderhut** [reply]
The webpage it wants to redirect me to and the end is just about the most confusing one I've read, so how do you fix this?? 
 
huggingface-cli download xai-org/grok-1 --repo-type model --include 'ckpt-0/*' --local-dir checkpoints --local-dir-use-symlinks False
Traceback (most recent call last):
  File "/Users/c/Library/Python/3.9/bin/huggingface-cli", line 5, in <module>
    from huggingface_hub.commands.huggingface_cli import main
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/huggingface_hub/commands/huggingface_cli.py", line 18, in <module>
    from huggingface_hub.commands.delete_cache import DeleteCacheCommand
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/huggingface_hub/commands/delete_cache.py", line 64, in <module>
    from ..utils import CachedRepoInfo, CachedRevisionInfo, HFCacheInfo, scan_cache_dir
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/huggingface_hub/utils/__init__.py", line 32, in <module>
    from ._errors import (
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/huggingface_hub/utils/_errors.py", line 3, in <module>
    from requests import HTTPError, Response
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/requests/__init__.py", line 43, in <module>
    import urllib3
  File "/Users/c/Library/Python/3.9/lib/python/site-packages/urllib3/__init__.py", line 38, in <module>
    raise ImportError(
ImportError: urllib3 v2.0 only supports OpenSSL 1.1.1+, currently the 'ssl' module is compiled with LibreSSL 2.8.3. See: https://github.com/urllib3/urllib3/issues/2168

**Comment by pythonmandev** [comment]


**Reply by AdaptiveStep** [reply]
we need more mirrors

**Comment by protonish** [comment]
the easiest way is
```
huggingface-cli download xai-org/grok-1 --repo-type model --include ckpt-0/* --local-dir checkpoints --local-dir-use-symlinks False
```

courtesy https://huggingface.co/xai-org/grok-1

**Comment by wingsum93** [comment]
@protonish Why the easiest way is to download from huggerface? Assuming the download speed is low (less then 100MB/s), computer is likely to reboot the whole period. 
Download in torrent allow us to pause and resume in easy way.

**Comment by traderhut** [comment]
there IS no ckpt-0 folder, so nothing in it for that command to use, the web page: https://github.com/xai-org/grok-1?tab=readme-ov-file#downloading-the-weights loops back to itself at the top saying to go there to get those files, and be sure to get that directory... OK...   Hocus-Pocus-Magically-APPEAR!!!   Nope, the files still aren't here... any CLUE how you actually do this?  Want to include a lik t a DIFFERENT page that actually tells me how to download the checkpoint files first?   The hugging-face thing won't work because the files aren't there, and nothing I can find says anything about how to get them!  (Hey, you could check them in to version control?)
